test/unit/plugins/connection_test.rb

```diff
@@ -14,6 +14,10 @@
       proc { connection.run_command('') }.must_raise Train::ClientError
     end
 
+    it 'provides a run_command_via_connection method' do
```

Comment:
I wonder if this should be `it 'raises an exception for run_command_via_connection'`, and then we add a `it 'provides a run_command_via_connection method'` check for each of our connection classes. Thoughts?